internal/controllers/inventory_controller.go

```diff
@@ -1082,6 +1094,69 @@ func (t *reconcilerTask) createAlarmServerClusterRoleBinding(ctx context.Context
 	return nil
 }
 
+// createAlertmanagerClusterRoleAndBinding creates the cluster role and binding needed for the Alertmanager to be able to
+// send alerts to the alarm server.
+func (t *reconcilerTask) createAlertmanagerClusterRoleAndBinding(ctx context.Context) error {
+	name := "alertmanager"
+
+	// Cluster Role
+	role := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf(
+				"%s-%s", t.object.Namespace, name,
+			),
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				NonResourceURLs: []string{
+					"/internal/v1/caas-alerts/alertmanager",
+				},
+				Verbs: []string{
+					// Expected to be post given that it is a nonResourceURLs, but it only works with create.
+					// Adding post in case something changes in the future. It doesn't hurt to have it.
+					"create",
+					"post",
+				},
+			},
+		},
+	}
+
+	if err := utils.CreateK8sCR(ctx, t.client, role, t.object, utils.UPDATE); err != nil {
+		return fmt.Errorf("failed to create Alertmanager cluster role: %w", err)
+	}
+
+	// Cluster Role Binding
+	binding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf(
+				"%s-%s",
+				t.object.Namespace, name,
+			),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name: fmt.Sprintf(
+				"%s-%s",
+				t.object.Namespace, name,
+			),
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Namespace: "open-cluster-management-observability",
```

Comment:
We should use a constant for this namespace name so it is not repeated in multiple places. ...same for the `alertmanager` name.

Reply:
updated, thanks